### DISCUSSION TITLE
Welcome to HP8753-Companion Discussions!

### DISCUSSION THREAD

**Discussion by VK2BEA** [VK2BEA/HP8753-Companion#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with HP8753 users. We hope that you:
  * Ask questions related to the program or the HP8753 network analyzer.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by uworgl** [comment]
Hello, the HP8753-Companion was recommended by a good friend (DK1VI) to me. I am not a Radio Amateur, but are running some RF Equipment in order to work on small RF projects. I have here a HP8753D including Test Set up to 6GHz, what is good enough for my private lab.
I have an Agilent 82357B running on Ubuntu 18.04 LTS. The control of the 8753 is possible using simple Python scripts so far.
The compilation of the HP8753-Companion was terminated with several failures. Can I ask for a little help here in order to get it running. If this is the right place, please let me know. Then I can provide more details.  🙂

**Reply by VK2BEA** [reply]
Yes .. but please start an "issue". Describe your Linux environment (gcc, Linux kernel, linux-gpib-devel version etc)  and the error messages you see.
I don't have an Ubuntu environment but I do have a Raspberry PI which is also Debian based. I haven't compiled on it for many months so something in the environment may have changed and broken things.
I'll reply to your issue and we can go from there..

Michael

**Reply by uworgl** [reply]
Hi Michael,
I have openend issue #7 
For any reply we might need to consider the time shift to Germany....   :-)

Volker

**Comment by Nico6824** [comment]
Hi Michael, I am new to the discussion. I hope this message finds you well. 

I came across your software and desperately wanted to give it a try on my HP8753ES (this is awesome). I have been going through the process step by step and I came across an issue - Perhaps this is something small or obvious: The issue occurs when proceeding to use the following instructions:

- $ make all
- $ sudo make install

I receive error message: 
  CCLD     hp8753
/usr/bin/ld: hp8753-USBTMC_interface.o: undefined reference to symbol 'aio_read@@GLIBC_2.17'
/usr/bin/ld: /lib/aarch64-linux-gnu/librt.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:488: hp8753] Error 1
make[1]: Leaving directory '/home/nico/HP8753-Companion/build/src'
make: *** [Makefile:420: all-recursive] Error 1

I cannot seem to find the reference to GLIB 2.17; all previous operations seem to have completed successfully but, I am unable to proceed and not quite sure what could be the cause. If you could share any insight or give me a reference to solving the issue, I would greatly appreciate it. 

Details to my version of GLIBC:
ldd (Debian GLIBC 2.31-13+rpt2+rpi1+deb11u11) 2.31

Kind regards,
Nico

**Comment by Nico6824** [comment]
Hi Michael, I hope you have been well. I just wanted to give an update and perhaps share some images which I saved in the HP8753-Companion software. People need to understand what a great job you did here. I did end up installing the NI 488.2 drivers on windows, with the version which was adapted for Windows. I have not had much time to play around with the raspberry PI and I decided to give the Windows version a try before continuing with Debian:  

- Currently I am able to take measurements ("get trace") using dual channel with auxiliary channel disabled - simultaneous 4 port measurements fail and freezes the analyzer (display 1x, 2x, 4x does not make any difference here). I did notice that running the program without being administrator does present a few issues as well. (one thing I noticed in particular: markers not being displayed in high resolution and marker cut-off in HPGL) 
- I did try to export the actual S1P/S2P files but, for some reason this interaction also freezes the analyzer (similar to measurement with aux channel) - The only resolution is to take the analyzer to Preset and restart the measurements.

The current adapter which I have is an NI GPIB-USB-HS, in case this would make any difference to the code that was written.  

Thank you very much for sharing the software - This is amazing stuff. 

<img width="3300" height="2550" alt="S22_2450_Cavity" src="https://github.com/user-attachments/assets/c92385b5-8877-4f44-a8e2-0ea046d59561" />
<img width="3300" height="2550" alt="S11_S22_Log" src="https://github.com/user-attachments/assets/09a67c12-b556-45ef-a4bc-dcda8901feb9" />
<img width="3300" height="2550" alt="Cavity_2450_S21_LOG_3500" src="https://github.com/user-attachments/assets/2ef8f6c5-e083-4bbf-9605-2b131718e520" />

**Comment by VK2BEA** [comment]
I'm glad you got it working (at least partially).
I have not seen the "Windows version" but I remember someone saying they would like to port it to Windows.

As to the crashes, it could be a number of things ....
Did you run the "Analyze Learn String" on the Options page?
The "E" model may be  (probably)  giving responses that I'm not expecting. Without a unit to test, it's hard to say ... and debugging it would be a challenge.
Also, the Windows port would bring in another set of unknowns.

**Reply by Nico6824** [reply]
I completely understand. I will definitely try the Debian route soon. It would be great to get the full functionality, as you intended on the ES. I did try the learn string option and it was able to verify firmware version. However, on the first try, the analyzer reacted quite strangely, and this had me a bit worried at first. after switching the unit off and back on again, everything was fine. Now that you mention it, I can't remember if the markers appeared after this check was initiated (initially the markers were not visible in high res).

Once I have some time, I will go through the setup step by step and give you some feedback. With all this said, the images have great quality - I am very happy, thanks again

**Reply by VK2BEA** [reply]
Yes, in Linux we can try to enable debug and get a good log of the communications. In that way I might be able to simulate the "ES" and understand what is failing.

**Comment by Nico6824** [comment]
Hi Michael, I will try this next - Once I manage to get it running, I will let you know. Thanks
